NSSAgent/NSSAgent.cs

```diff
@@ -1070,10 +1101,37 @@ public IQueryable<VariableType> GetVariables()
             // (var, unit) => new VariableType { ID = var.ID, Code = var.Code, Description = var.Description, UnitTypeID = unit.UnitTypeID });
             return this.Select<VariableType>();
         }
+        public IQueryable<object> GetVariablesWithUnits()
+        {
+            IQueryable<Variable> unitTypes = this.Select<Variable>().Where(x => x.Comments == "Default unit");
+            var obj = this.Select<VariableType>().Join(unitTypes, var => var.ID, unit => unit.VariableTypeID, (var, unit) => new
+            {
+                ID = var.ID,
+                Description = var.Description,
+                Code = var.Code,
+                Name = var.Name,
+                UnitTypeID = unit.UnitTypeID
+            }).Distinct().OrderBy(x => x.ID);
+
+            return obj;
+        }
         public Task<VariableType> GetVariable(Int32 ID)
         {
             return this.Find<VariableType>(ID);
         }
+        public object GetVariableWithUnits(Int32 ID)
+        {
+            IQueryable<Variable> unitTypes = this.Select<Variable>();
```

Comment:
I don't see a check for Comments == "Default unit" here. If we have variable types that are used in an equation, we might get the wrong variable returned.

Reply:
Forgot to carry that over from the other VariablesWithUnits method.